### PR TITLE
fix: prevent recursive Codex notify chaining

### PR DESCRIPTION
**Pull request by Wayneoween**
## Summary

Prevents an infinite Codex notification loop when Sky Computer Use already delegates to TokenTracker through `--previous-notify`.

TokenTracker's serve-time repair captured the Sky wrapper and replaced it with its own notifier. The generated handler then called Sky, whose previous notifier pointed back to TokenTracker. Each callback spawned another `SkyComputerUseClient` and `node` process.

On macOS, that process storm also drove up `syspolicyd`, `XProtect` and `diagnosticd` CPU usage until the machine was fully saturated.

This change leaves an existing Sky-to-TokenTracker wrapper alone. The generated notifier also checks nested JSON callbacks for self-references, which protects machines that already have a bad saved notification chain.

## Scope

- [x] CLI (`src/`)
- [ ] Dashboard (`dashboard/`)
- [ ] macOS app (`TokenTrackerBar/`)
- [ ] Windows app (`TokenTrackerWin/`)
- [ ] Docs / CI / config

## Checklist

- [x] `npm test` passes
- [x] New user-facing strings go through `dashboard/src/content/copy.csv` (no user-facing strings added)
- [x] Commits follow conventional style (`feat:` / `fix:` / `refactor:` / `docs:` / `chore:` / `test:` / `ci:`)
- [x] PR description explains *why*, not just *what*

---

<details>
<summary><strong>Risk layer addendum</strong> — expand if this PR touches any trigger below</summary>

### Risk layer triggers

- [ ] Public exposure / share links / unauthenticated access
- [ ] Auth / session / token handling
- [x] Cross-endpoint invariants or shared logic
- [ ] External gateway / environment constraints

### Rules / invariants

- A notification chain must never invoke TokenTracker again.
- Self-detection must work when the callback is stored as a JSON string.
- A Sky wrapper that already delegates to TokenTracker must remain unchanged.
- Existing non-recursive notification commands must still run.

### Boundary matrix (list at least 3)

| Configuration | Expected behavior |
|---|---|
| Plain Sky notifier | TokenTracker may preserve and chain it |
| Sky with TokenTracker in `--previous-notify` | Leave the active notifier unchanged |
| Stored notifier with a nested TokenTracker callback | Skip the stored notifier |
| Unrelated executable notifier | Continue chaining it |

### Public exposure checklist (if applicable)

- [ ] Public access rules defined (share token required, non-JWT handling, 401 behavior)
- [ ] Exposed fields explicitly listed and verified
- [ ] Avatar/image policy defined
- [ ] Regression tests cover invalid link and auth fallback

</details>

<details>
<summary><strong>Codex review context</strong> — fill when requesting <code>@codex</code> review</summary>

- **Delta since last Codex review:** Initial review request. Commit `6eb2338` adds nested self-reference detection and preserves an existing Sky-to-TokenTracker wrapper.
- **Intended behavior / invariants:** Notification chains must never call TokenTracker recursively. Existing non-recursive notifiers must keep working, including plain Sky wrappers and unrelated executables.
- **Edge cases covered:** TokenTracker nested inside a JSON-encoded `--previous-notify`; serve-time repair encountering a Sky wrapper that already points back to TokenTracker.
- **Tests run (command + result):** `npm test`, 1,350 passed and 0 failed. The 2 regression tests failed before the fix and passed afterward.
- **Known gaps / out of scope:** No UI, public-access, authentication, or platform-specific app behavior changed.

### Most likely regression surface

- Codex notification chaining for existing Sky Computer Use wrappers and custom executable notifiers.

### Verification method (choose at least one)

- Added a regression test using the exact nested `--previous-notify` shape that caused the loop.
- Added a serve-time repair test for an existing Sky-to-TokenTracker wrapper.
- Both tests failed before the fix and passed afterward.
- Ran the full test suite: 1,350 passed and 0 failed.

### Uncovered scope

- Manual verification on Windows and Linux. The changed code is shared CLI notification handling, and the automated suite covers it.

</details>

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved handling of nested notification handlers, including SkyComputerUse integrations.
  * Prevented duplicate wrapping when a TokenTracker notification is already nested.
  * Improved detection of self-referencing notification commands.
  * Preserved existing configuration and removed unnecessary backup requirements during repair.

### CODE REVIEW

File: test/init-uninstall.test.js [hunk 1]
```diff
@@ -107,6 +107,37 @@ test("notify handler chains executable original notify commands and skips stale
   }
 });
 
+test("notify handler skips an original notify that nests itself", async () => {
+  const tmp = await fs.mkdtemp(path.join(os.tmpdir(), "tokentracker-notify-chain-"));
+  try {
+    const trackerDir = path.join(tmp, "tracker");
+    await fs.mkdir(trackerDir, { recursive: true });
+    const notifyPath = path.join(await fs.realpath(trackerDir), "notify.cjs");
```
[the remark this answers is not in the document]
Comment (Wayneoween) [reply test/init-uninstall.test.js:115]: Although `path.resolve(__filename)` doesn't resolve symlinks itself, Node canonicalizes the entry script before assigning `__filename`. For example, executing a script through `/tmp/...` produces `/private/tmp/...` as `__filename`. The same applies to `os.tmpdir()`, which returns `/var/folders/...` while the executed handler sees `/private/var/folders/...`.

Using `fs.realpath(trackerDir)` therefore makes the nested callback path match the generated handler's `selfPath`. Removing it would make those paths diverge on macOS.

I tested both exact versions on macOS.

CodeRabbit’s version:

```
const notifyPath = path.join(trackerDir, "notify.cjs");
```

Produces paths like:

```
nested callback: /var/folders/.../tracker/notify.cjs
handler selfPath: /private/var/folders/.../tracker/notify.cjs
```

Test result:

```
'ran' !== null
1 failed
```

The handler misses the self-reference and executes Sky. Meanwhile our version:

```
const notifyPath = path.join(
  await fs.realpath(trackerDir),
  "notify.cjs",
);
```

Produces:

```
nested callback: /private/var/folders/.../tracker/notify.cjs
handler selfPath: /private/var/folders/.../tracker/notify.cjs
```
 
Result:
```
1 passed
0 failed
```
